src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerHandler.cs

```diff
@@ -219,14 +224,14 @@ internal class JwtBearerHandler : AuthenticationHandler<JwtBearerOptions>
                     // Only add a comma after the first param, if any
                     builder.Append(',');
                 }
-                builder.Append(" error=\"invalid_token\", error_description=\"");
-                builder.Append(errorDescription);
+                builder.Append($" error=\"{eventContext.Error ?? "invalid_token"}\", error_description=\"");
+                builder.Append(eventContext.ErrorDescription ?? "An error occurred when validating the access token.");
                 builder.Append('\"');
 
                 Response.Headers.Append(HeaderNames.WWWAuthenticate, builder.ToString());
             }
 
-            return false;
+            return true;
```

Comment:
This should stay `false`. It doesn't just mean "did I handle this?", but rather "should I exclude others from handling this?" See https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNetCore.Authentication/AuthenticationHandler.cs#L334
The redirecting providers are all exclusive and would return true, but providers like Basic, Bearer, etc. can all contribute challenges to the same response in an additive fashion.

I'll take care of this and get this merged today.